name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/EurostatHICPQuoteFeed.java

```diff
@@ -0,0 +1,234 @@
+    {
+        LocalDate quoteStartDate = LocalDate.MIN;
+
+        if (!security.getPrices().isEmpty())
+            quoteStartDate = security.getPrices().get(security.getPrices().size() - 1).getDate();
+
+        List<LatestSecurityPrice> prices = getHistoricalQuotes(security, quoteStartDate, errors);
+
+        boolean isUpdated = false;
+        for (SecurityPrice p : prices)
+        {
+            if (p.getDate().isBefore(LocalDate.now()))
+            {
+                boolean isAdded = security.addPrice(p);
+                isUpdated = isUpdated || isAdded;
+            }
+        }
+        return isUpdated;
+    }
+
+    @Override
+    public List<LatestSecurityPrice> getHistoricalQuotes(Security security, LocalDate start, List<Exception> errors)
+    {
+        security.setCalendar(TradeCalendar.EMPTY_CODE);
+
+        if (security.getTickerSymbol() == null)
+        {
+            errors.add(new IOException(MessageFormat.format(Messages.MsgMissingTickerSymbol, security.getName())));
+            return Collections.emptyList();
+        }
+
+        try (CloseableHttpClient client = HttpClients.createSystem())
+        {
+            URIBuilder uriBuilder = new URIBuilder().setScheme("http").setHost(EUROSTAT_HOST) //$NON-NLS-1$
+                            .setPath(EUROSTAT_PAGE);
+            uriBuilder.addParameter("filterNonGeo", "1"); //$NON-NLS-1$ //$NON-NLS-2$
+            uriBuilder.addParameter("precision", "1"); //$NON-NLS-1$ //$NON-NLS-2$
+            uriBuilder.addParameter("geo", security.getTickerSymbol().toUpperCase()); //$NON-NLS-1$
+            uriBuilder.addParameter("unit", "I15"); //$NON-NLS-1$ //$NON-NLS-2$
+            uriBuilder.addParameter("unitLabel", "code"); //$NON-NLS-1$ //$NON-NLS-2$
+            uriBuilder.addParameter("coicop", "CP00"); //$NON-NLS-1$ //$NON-NLS-2$
+            uriBuilder.addParameter("groupedIndicators", "1"); //$NON-NLS-1$ //$NON-NLS-2$
+            uriBuilder.addParameter("shortLabel", "1"); //$NON-NLS-1$ //$NON-NLS-2$
+
+            URL objectURL = uriBuilder.build().toURL();
+            try (CloseableHttpResponse response = client.execute(new HttpGet(objectURL.toString())))
+            {
+                if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK)
+                    throw new IOException(objectURL.toString() + " --> " + response.getStatusLine().getStatusCode()); //$NON-NLS-1$
+
+                String body = EntityUtils.toString(response.getEntity());
+                JSONObject eurostatJSON = (JSONObject) JSONValue.parse(body);
+                JSONObject eurostatValue = (JSONObject) eurostatJSON.get("value"); //$NON-NLS-1$
+                JSONObject eurostatDimension = (JSONObject) eurostatJSON.get("dimension"); //$NON-NLS-1$
+                JSONObject eurostatTime = (JSONObject) eurostatDimension.get("time"); //$NON-NLS-1$
+                JSONObject eurostatCategory = (JSONObject) eurostatTime.get("category"); //$NON-NLS-1$
```

Comment:
Da die Objekte Category, Dimension, Time nicht weiter gebraucht werden, muss man sie auch nicht umbedingt in ein Objekt einlesen, oder?

Reply:
Um an die einzelnen Perioden zu kommen muss ich mich hierarchisch durch 4 Ebenen graben. Aber ich muss zugeben das ich nicht ausprobiert habe ob sich die GET Anweisungen aneinander Reihen lassen, alla `.get (Category).get(Dimension).get(Time).get(Index). Das probiere ich aus.

Reply:
Äh, Sorry, klar. Da habe ich falsch gelegen. Macht schon Sinn so. Und ist dann auch lesbar.